Adapt SupportInformationRetriever for Exasol v8

Integration tests for `SupportInformationRetriever` fail with Exasol v8 with the following error:

```
Error:  com.exasol.support.SupportInformationRetrieverIT.testArchiveProducedWhenExitWithErrorTriggerIsConfigured(Path) -- Time elapsed: 35.32 s <<< FAILURE!
java.lang.AssertionError: Unable to check entry 'sysinfo.txt' in archive '/tmp/junit13726521362876133579/exacluster_debuginfo_2024_04_16-15_14_21.tar.gz'. Cause: java.io.EOFException
	at com.exasol.support.SupportInformationRetrieverIT.assertTarArchiveContainsEntry(SupportInformationRetrieverIT.java:70)
	at com.exasol.support.SupportInformationRetrieverIT.testArchiveProducedWhenExitWithErrorTriggerIsConfigured(SupportInformationRetrieverIT.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Caused by: java.io.EOFException
	at java.base/java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:269)
	at java.base/java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:259)
	at java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:165)
	at java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:80)
	at java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:92)
	at com.exasol.support.SupportInformationRetrieverIT.assertTarArchiveContainsEntry(SupportInformationRetrieverIT.java:58)
	... 72 more
```

Log contains the following error message:
```
 2024-04-16 13:14:21.680 [SEVERE ] E-ETC-2: Container exiting with EXIT_ERROR. Monitoring is set to EXIT_ERROR. Unable to create support archive '/tmp/junit13726521362876133579'.
Cause: 'Traceback (most recent call last):
  File "/opt/exasol/cos-8.45.0/bin/exasupport", line 8, in <module>
    sys.exit(main())
  File "/opt/exasol/cos-8.45.0/lib/python3.10/site-packages/libconfd/cli/exasupport.py", line 25, in main
    _, act_out = support.downloadDebugInfo(request_logs, request_bts, nodes, args.outfile)
  File "/opt/exasol/cos-8.45.0/lib/python3.10/site-packages/libconfd/common/support.py", line 469, in downloadDebugInfo
    raise RuntimeError('Please specify debug information to download.')
RuntimeError: Please specify debug information to download.
' 
```

--- a/doc/changes/changes_7.1.2.md
+++ b/doc/changes/changes_7.1.2.md
@@ -1,4 +1,4 @@
-# Test Containers for Exasol on Docker 7.1.2, released 2024-??-??
+# Test Containers for Exasol on Docker 7.1.2, released 2024-10-22
 
 Code name: Allow withExposedPorts
 
@@ -8,11 +8,14 @@ This release fixes an issue that prevented the container startup when using the 
 The installation of ScriptLanguageContainers requires the BucketFS internal port to be available, but this shouldn't 
 be the case when no SLC has been requested to be installed.
 
-It also bumps default docker image to 7.1.30.
+It also adapts the support of information retrieval at exit to work with Exasol v8.
+
+The default docker image was bumped to 7.1.30.
 
 ## Bugfix
 
 * #257: Container startup fails when using withExposedPorts()
+* #254: Adapt SupportInformationRetriever for Exasol v8
 
 ## Dependency Updates
 

--- a/src/main/java/com/exasol/support/SupportInformationRetriever.java
+++ b/src/main/java/com/exasol/support/SupportInformationRetriever.java
@@ -26,7 +26,7 @@ public class SupportInformationRetriever {
     /** Name of the property that allows overriding the monitored container exit */
     public static final String MONITORED_EXIT_PROPERTY = "com.exasol.containers.monitored_exit";
     static final String SUPPORT_ARCHIVE_PREFIX = "exacluster_debuginfo_";
-    private static final String EXASUPPORT_EXECUTABLE = "exasupport";
+    private static final String[] EXASUPPORT_COMMAND = {"exasupport", "-d", "0"};
     private static final Logger LOGGER = LoggerFactory.getLogger(SupportInformationRetriever.class);
     private static final String MAPPED_HOST_DIRECTORY = "/exa/tmp/support";
     private static final ExitType DEFAULT_MONITORED_EXIT_TYPE = ExitType.EXIT_NONE;
@@ -95,7 +95,7 @@ public class SupportInformationRetriever {
     @SuppressWarnings("java:S112")
     private void createArchiveBundle(final ExitType exitType) {
         try {
-            final ExecResult result = this.container.execInContainer(EXASUPPORT_EXECUTABLE);
+            final ExecResult result = this.container.execInContainer(EXASUPPORT_COMMAND);
             if (result.getExitCode() == ExitCode.OK) {
                 final String filename = extractFilenameFromConsoleMessage(result);
                 final String hostPath = getHostPath(filename);

--- a/src/test/java/com/exasol/support/SupportInformationRetrieverIT.java
+++ b/src/test/java/com/exasol/support/SupportInformationRetrieverIT.java
@@ -32,13 +32,6 @@ class SupportInformationRetrieverIT {
     private static final Logger LOGGER = LoggerFactory.getLogger(SupportInformationRetrieverIT.class);
     private static final String SYSINFO_FILENAME = "sysinfo.txt";
 
-    @BeforeAll
-    static void assumeSupportPackageSupported() {
-        // These tests don't work with Exasol v8. We will fix this in
-        // https://github.com/exasol/exasol-testcontainers/issues/254
-        ExasolContainerAssumptions.assumeDockerDbVersionNotOverriddenToBelowExasolEight();
-    }
-
     // [itest->dsn~configure-support-information-retriever-via-api~1]
     @Test
     void testWriteSupportBundleOnExit(@TempDir final Path tempDir) {

